### ISSUE TITLE
Make importance configurable

### ISSUE THREAD

**Issue by jancp** [bram85/topydo#180]
Hello!

I really like the importance feature of topydo. However, I don't like the way the importance is calculated too much. I would appreciate if you could be able to configure importance to your needs in a config file.

For example, like in the description of the feature: `"2 + p + s + d"` where _p_, _s_, and _d_ are variables. (Quite frankly I believe the [description](http://www.toodledo.com/info/help.php?sel=53) of the feature in Toodledo is a typo. The `2 +` basically does nothing -- it probably should have been `2 *`. But that's not the issue here.)

That way you could adapt the importance to your needs. For example like [Tasks for Android](https://github.com/tasks/tasks) does it: `"p * 2 * now + ((2 * now) if duedate == 0 else duedate)"`

Or something else entirely, at the user's choice.

Thank you for considering this request and keep up the great work!

Cheers,

Jan

**Comment by bram85** [comment]
I think it's quite advanced, not a must have on my list. We should come up with a way to parse expressions like these, easier said than done (especially if you allow parentheses to group). Proper parsing has also turned up in discussions around using the OR operator in `ls` (#177), so we could apply the same technology here (i.e. parser tech).

By the way, why do you think `2 +` is a typo? I checked with Toodledo and the calculation is really done according to the documentation.

**Comment by jancp** [comment]
I agree this is quite advanced, but it would still be a nice improvement. :)

> By the way, why do you think 2 + is a typo?

Because it makes no difference to the calculation as a whole. You just add two to everything regardless of the other values.  If you have three tasks where _P + S + D_ amount to an importance of 0, 1, and 2 respectively, you end up with 2, 3, and 4. One task is still ahead of another by 1. It does not influence the relative importance of the tasks in any way.
